### PR TITLE
don't draw inactive drawables, implement flip functions for drawables

### LINKED ISSUE #50: Drawable's still draw when the inactive flag is set

### LINKED ISSUE #51: Implement functions to flip drawables on the X and Y axes

### PR DESCRIPTION
**Pull request by tlein**
Resolves #50 
Resolves #51

### CODE REVIEW

File: Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.cpp [hunk 1]
```diff
@@ -54,3 +57,32 @@ Drawable * Drawable::FindDrawable(const std::string &key)
     return nullptr;
 }
 
+void Drawable::FlipX()
```
Comment (jeffswenson) [inline Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.cpp]: You should remove FlipX and FlipY. If there is an idempotent way to implement an API (like SetXDirection), that should be the only API provided. Idempotent APIs are generally better behaved.

Comment (tlein) [reply Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.cpp]: fixed

File: Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.cpp [hunk 2]
```diff
@@ -54,3 +57,22 @@ Drawable * Drawable::FindDrawable(const std::string &key)
...
 }
 
+void Drawable::SetXDirection(int leftOrRightSignum)
+{
+    if (_scale.x < 0)
+    {
+        _scale.x *= -1;
+    }
+
+    _scale.x *= leftOrRightSignum;
```
Comment (jeffswenson) [inline Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.cpp]: _scale.x = std::abs(_scale.x) * leftOrRightSignum;

File: Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.hpp [hunk 3]
```diff
@@ -86,6 +86,18 @@ class Drawable
...
+         * Sets the direction of the drawable in the x axis.
+         * @param leftOrRightSignum -1 if the drawable should be facing left, 1 if right.
+         */
+        void SetXDirection(int leftOrRightSignum);
+
+        /**
+         * Sets the direction of the drawable in the y axis.
+         * @param leftOrRightSignum -1 if the drawable should be facing down, 1 if up.
+         */
+        void SetYDirection(int leftOrRightSignum);
```
Comment (jeffswenson) [inline Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.hpp]: You should rename this parameter up or down.

Comment (tlein) [reply Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.hpp]: fixed

File: Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.cpp (hunk 2 above)
Comment (tlein) [reply Engine/Src/Ancona/Core2D/Systems/Drawable/Drawable.cpp]: fixed

**Pull request merged**